.cursor/rules/code-style.mdc

```diff
@@ -0,0 +1,469 @@
+- Use `# type: ignore` comments only when absolutely necessary and document why
+- Prefer `# type: ignore[error-code]` over blanket ignores
+- Run MyPy locally before committing: `mypy src/`
+
+### Common Typing Patterns
+
+1. **Async functions**:
+   ```python
+   async def fetch_data(url: str) -> Dict[str, Any]:
+       pass
+   ```
+
+2. **Class methods**:
+   ```python
+   class MyClass:
+       def instance_method(self, param: str) -> int:
+           pass
+       
+       @classmethod
+       def class_method(cls, param: str) -> "MyClass":
+           pass
+       
+       @staticmethod
+       def static_method(param: str) -> bool:
+           pass
+   ```
+
+3. **Type aliases for complex types**:
+   ```python
+   from typing import TypeAlias
+   
+   ConfigDict: TypeAlias = Dict[str, Union[str, int, bool]]
+   ResultList: TypeAlias = List[Dict[str, Any]]
+   ```
+
+## OS Operations Must Use Adaptors
+
+**CRITICAL**: Never use OS operations directly in `src/` code. Always use adaptors for better testability and maintainability.
+
+### OS Adaptor Requirements
+
+1. **Use existing adaptors in `src/dd_license_attribution/adaptors/`**:
+   ```python
+   # ✅ Correct
+   from dd_license_attribution.adaptors.os import OSAdaptor
+   
+   class FileProcessor:
+       def __init__(self, os_adaptor: OSAdaptor) -> None:
+           self.os_adaptor = os_adaptor
+       
+       def process_file(self, path: str) -> str:
+           if self.os_adaptor.path_exists(path):
+               return self.os_adaptor.read_file(path)
+           return ""
+   
+   # ❌ Incorrect - direct OS usage
+   import os
+   
+   class FileProcessor:
+       def process_file(self, path: str) -> str:
+           if os.path.exists(path):  # Direct OS usage
+               with open(path) as f:  # Direct OS usage
+                   return f.read()
+           return ""
+   ```
+
+2. **Create new adaptors when needed**:
+   ```python
+   # ✅ Correct - new adaptor
+   from abc import ABC, abstractmethod
+   from typing import Protocol
+   
+   class NetworkAdaptor(Protocol):
+       def make_request(self, url: str) -> str: ...
+   
+   class RealNetworkAdaptor:
+       def make_request(self, url: str) -> str:
+           import requests
+           return requests.get(url).text
+   
+   class MockNetworkAdaptor:
+       def make_request(self, url: str) -> str:
+           return "mocked_response"
+   ```
+
+3. **Easy mocking pattern**:
+   ```python
+   # ✅ Correct - easy to mock
+   class Service:
+       def __init__(self) -> None:
+           self.os_adaptor = OSAdaptor()
+           self.network_adaptor = NetworkAdaptor()
+   
+   # Easy to mock in tests
+   with patch('dd_license_attribution.adaptors.os.OSAdaptor') as mock_os:
+       mock_os.return_value.read_file.return_value = "test content"
+       service = Service()
+       result = service.process_file("test.txt")
+   ```
+
+### Forbidden Direct Imports in `src/`
+
+- ❌ `import os`
+- ❌ `import sys`
```

Comment:
we currently import `sys` in the main_cli

Reply:
We only using it for checking platform at the entry point, I think its fine to have exceptions, specially in main. I will quote Captain Barbosa about this "The code is more what you'd call 'guidelines' than actual rules" being the code this file :)